### PR TITLE
all: remove legacy health check handlers

### LINKED ISSUE #977: appengine/flexible: use split health checks
From deploying helloworld:

> GAE Flex legacy health check is deprecated and its support will end soon. Please migrate to updated health checks. See https://cloud.google.com/appengine/docs/flexible/python/reference/app-yaml#updated_health_checks for instructions.

### PR DESCRIPTION
**Pull request by tbpg**
Legacy health checks migrated to split health checks, which are now the
default. Split health checks are not sent to apps by default, so these
handlers are not working as intended.

None of the app.yaml files have health checks configured, so I chose not
to add new split handlers & configuration. I'm open to adding them
if we decide we should.

Fixes #977.

### CODE REVIEW

File: appengine_flexible/helloworld/helloworld.go [hunk 1]
```diff
@@ -42,7 +41,3 @@ func handle(w http.ResponseWriter, r *http.Request) {
 	}
 	fmt.Fprint(w, "Hello world!")
 }
-
-func healthCheckHandler(w http.ResponseWriter, r *http.Request) {
-	fmt.Fprint(w, "ok")
-}
```
Comment (grayside) [inline appengine_flexible/helloworld/helloworld.go:48]: Used in the quickstart, healthchecks can be skipped.

File: appengine_flexible/websockets/main.go [hunk 2]
```diff
@@ -70,9 +68,4 @@ func socketHandler(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
-// healthCheckHandler is used by App Engine Flex to check instance health.
-func healthCheckHandler(w http.ResponseWriter, r *http.Request) {
-	fmt.Fprint(w, "ok")
-}
-
```
Comment (grayside) [inline appengine_flexible/websockets/main.go:77]: Healthchecks are noted in the [docs for websockets](https://cloud.google.com/appengine/docs/flexible/go/using-websockets-and-session-affinity?hl=en) as part of how the platform relocates sessions to preserve connections using session affinity. Migrating to split healthchecks seems valuable.

File: getting-started/bookshelf/main.go [hunk 3]
```diff
@@ -100,13 +100,6 @@ func (b *Bookshelf) registerHandlers() {
...
 
-	// Respond to App Engine and Compute Engine health checks.
-	// Indicate the server is healthy.
-	r.Methods("GET").Path("/_ah/health").HandlerFunc(
-		func(w http.ResponseWriter, r *http.Request) {
-			w.Write([]byte("ok"))
-		})
-
 	r.Methods("GET").Path("/logs").Handler(appHandler(b.sendLog))
 	r.Methods("GET").Path("/errors").Handler(appHandler(b.sendError))
```
Comment (grayside) [inline getting-started/bookshelf/main.go:105]: I'm getting a bit turned around between quickstart, getting started, and bookshelf, but none of these entry-level tutorials seem to need healthchecks.

File: appengine_flexible/websockets/main.go (hunk 2 above)
Comment (tbpg) [reply appengine_flexible/websockets/main.go:77]: That doc mentions "If the target instance fails health checks, App Engine moves the session to a healthy instance." That seems very reasonable.

https://cloud.google.com/appengine/docs/flexible/custom-runtimes/configuring-your-app-with-app-yaml#updated_health_checks mentions health checks are on by default and only need to be customized if you want to. So, I'm not sure it's valuable to add health checks here. Maybe we should update the websocket doc to link to the health check docs?

Comment (grayside) [reply appengine_flexible/websockets/main.go:77]: Agreed, let's remove this and cross-link the docs. I'll send a CL to the docs now.

**Pull request merged**